activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter-hal-rest-api/src/test/java/org/activiti/definition/ProcessDefinitionIT.java

```diff
@@ -89,4 +92,43 @@ public void shouldReturnProcessDefinitionById() throws Exception {
         assertThat(entity.getBody()).isNotNull();
         assertThat(entity.getBody().getId()).isEqualTo(aProcessDefinition.getId());
     }
+    
+    @Test
+    public void shouldReturnProcessDefinitionMetadata() throws Exception {
+    	//given
+        ParameterizedTypeReference<ProcessDefinition> responseType = new ParameterizedTypeReference<ProcessDefinition>() {
+        };
+
+        ResponseEntity<PagedResources<ProcessDefinition>> processDefinitionsEntity = getProcessDefinitions();
+        assertThat(processDefinitionsEntity).isNotNull();
+        assertThat(processDefinitionsEntity.getBody()).isNotNull();
+        assertThat(processDefinitionsEntity.getBody().getContent()).isNotEmpty();
+        ProcessDefinition aProcessDefinition = null;
+        	
+        Iterator<ProcessDefinition> it = processDefinitionsEntity.getBody().getContent().iterator();
+        do
+        {
+        	aProcessDefinition = it.next();
+        }
+        while(!aProcessDefinition.getName().equals(PROCESS_WITH_VARIABLES_2));
+        
+        //when
+        ResponseEntity<ProcessDefinition> entity = restTemplate.exchange(PROCESS_DEFINITIONS_URL + aProcessDefinition.getId() + "/meta",
+                                                                              HttpMethod.GET,
+                                                                              null,
+                                                                              responseType);
+        //then
+        assertThat(entity).isNotNull();
+        assertThat(entity.getBody()).isNotNull();
+        assertThat(entity.getBody().getVariables()).isNotNull();
+        assertThat(entity.getBody().getVariables().size()).isEqualTo(3);
```

Comment:
you can use `assertThat(entity.getBody().getVariables()).hasSize(3)`, then no check check if variables are not null (line 123). This will be done by `hasSize` method. Same for all others size checks.